pipeline/src/pipeline.rs

```diff
@@ -398,6 +398,26 @@ impl<T: FieldElement> Pipeline<T> {
         }
     }
 
+    pub fn from_maybe_optimized_file(self, file: PathBuf) -> Self {
+        if file.extension().unwrap() == "pilo" {
+            self.from_optimized_pil_file(file)
+        } else {
+            self.from_file(file)
+        }
+    }
+
+    pub fn from_optimized_pil_file(self, pil_file: PathBuf) -> Self {
```

Comment:
Sorry for being pedantic, but since we're changing names...

```suggestion
    pub fn from_pil_object(self, pil_file: PathBuf) -> Self {
```

Reply:
Haha no problem, I'm on it

Reply:
I should maybe also change the `from_maybe_optimized_file` to `from_maybe_pil_object` ?